pkg/providers/cloudflare/dnsrecord.go

```
@@ -301,12 +301,18 @@ func deleteRecord(ctx context.Context, client *cloudflare.API, zone string, fqdn
 	records := append(txtRecords, aRecords...)
 
 	for _, record := range records {
-		err := client.DeleteDNSRecord(ctx, zoneID, record.ID)
-		if err != nil {
-			metrics.ExecErrInc(err.Error())
-			return false, err
+		// validate record to be deleted. Only records with name same as the fqdn input and type `TXT` or `A` are allowed to be deleted
+		if record.Name == fqdn && (record.Type == "TXT" || record.Type == "A") {
+			err := client.DeleteDNSRecord(ctx, zoneID, record.ID)
+			if err != nil {
+				metrics.ExecErrInc(err.Error())
+				return false, err
+			}
+			logger.Info("Deleted DNS record", "zone", zone, "record", record.Name, "type", record.Type)
+		} else {
+			errr := fmt.Errorf("deleteRecord() wants to delete wrong record. Record Name: %v Record Type: %v", record.Name, record.Type)
```

Why do we have a new `errr` variable here? can't we use `err` ? 🤔

---

you mean why we create a new error? or why we name it `errr`?

---

Yeah sorry I meant why not re-use the `err` variable ?

---

the err variable gets a value once the [client.DeleteDNSRecord](https://github.com/gathertown/casper-3/pull/41/files#diff-021a0ca49e2dc5a2e2dcaf99c70d1496d0f8c3c6616722cc039f05d219531e70R306) being called. in our case,`client.DeleteDNSRecord` is not being called, so I would like to create and return an error, instead of returning just `false, nil`. Does that make sense?

---

I does but I am asking a different thing :)

This
```
errr := fmt.Errorf("deleteRecord() wants to delete wrong record. Record Name: %v Record Type: %v", record.Name, record.Type)
```
uses a `errr` variable to get the error. What if we write this as
```
err := fmt.Errorf("deleteRecord() wants to delete wrong record. Record Name: %v Record Type: %v", record.Name, record.Type)
```
so we use `err` as variable again? Right now it seems a bit confusing to have both `err` and `errr` to capture errors. But maybe this is a common practice so feel free to ignore me :)

---

ah sorry, I misunderstood the question. the `errr` can be named `err` again for sure. Actually, let me do that, since that looks:
1. better
2. more close to best/common practices

---

done: https://github.com/gathertown/casper-3/commit/bff4c0e472887323dbd604eeadb43e35f91893f5